presto-parser/src/main/java/com/facebook/presto/sql/parser/AstBuilder.java

```diff
@@ -1457,6 +1462,93 @@ protected Node aggregateResult(Node aggregate, Node nextResult)
         throw new UnsupportedOperationException("not yet implemented");
     }
 
+    private static String decodeString(SqlBaseParser.UnicodeStringContext context)
+    {
+        final char delimiter;
+        if (context.UESCAPE() != null) {
+            String delimiterString = context.BASIC_STRING().getText();
+            if (delimiterString.length() != 3) {
+                throw parseError("Invalid escaped delimiter: " + delimiterString, context);
+            }
+            delimiter = unquote(delimiterString).charAt(0);
+            if (isHexDigit(delimiter) || delimiter == '"' || delimiter == '+' || delimiter == ' ') {
```

Comment:
The specification says "whitespace", so use `Character.isWhiteSpace()`. Also validate that it's not a surrogate.